### ISSUE TITLE
createStorage() could return nil storage

### ISSUE THREAD

**Issue by hongchaodeng** [coreos/etcd-operator#505]
Test failed:
```
02:37:09 time="2016-12-23T02:35:05Z" level=info msg="Start reconciling" cluster-name=test-etcd-qxcve pkg=cluster 
02:37:09 time="2016-12-23T02:35:05Z" level=info msg="Finish reconciling" cluster-name=test-etcd-qxcve pkg=cluster 
02:37:09 time="2016-12-23T02:35:09Z" level=info msg="deleting cluster" cluster-name=test-etcd-r3kvz pkg=cluster 
02:37:09 time="2016-12-23T02:35:09Z" level=info msg="backup replica set and service deleted" cluster-name=test-etcd-r3kvz pkg=cluster 
02:37:09 panic: runtime error: invalid memory address or nil pointer dereference
02:37:09 [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x6ff41e]
02:37:09 
02:37:09 goroutine 54 [running]:
02:37:09 panic(0x19b9560, 0xc420012030)
02:37:09 	/usr/local/go/src/runtime/panic.go:500 +0x1a1
02:37:09 github.com/coreos/etcd-operator/pkg/cluster.(*backupManager).cleanup(0xc4203bb170, 0x0, 0x0)
02:37:09 	/home/jenkins/workspace/build-etcd-operator-master-176/gopath/src/github.com/coreos/etcd-operator/pkg/cluster/backup_manager.go:100 +0x18e
02:37:09 github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).delete(0xc4203b8b40)
02:37:09 	/home/jenkins/workspace/build-etcd-operator-master-176/gopath/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:355 +0x38d
02:37:09 github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).run.func1(0xc420295d4f, 0xc4203b8b40, 0xc42017d670)
02:37:09 	/home/jenkins/workspace/build-etcd-operator-master-176/gopath/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:181 +0x7c
02:37:09 github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).run(0xc4203b8b40, 0xc4203e3ce0, 0xc42017d670)
02:37:09 	/home/jenkins/workspace/build-etcd-operator-master-176/gopath/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:199 +0x38e
02:37:09 created by github.com/coreos/etcd-operator/pkg/cluster.new
02:37:09 	/home/jenkins/workspace/build-etcd-operator-master-176/gopath/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:143 +0x508
```

**Comment by hongchaodeng** [comment]
The root cause is that we deploy the wrong aws key.
NewS3Storage() returned error but the error slipped away:
https://github.com/coreos/etcd-operator/blob/dac0be35be238af5f51e79055ea2f76be882edd3/pkg/cluster/backup_manager.go#L69

**Comment by hongchaodeng** [comment]
Fixed in #506
